ui/src/components/datatypeselecter.vue

```diff
@@ -65,6 +66,18 @@ export default {
             this.$emit('input', value);
         }
     },
+    
+    methods: {
+        make_color: function(name) {
```

Comment:
Why is it necessary to set color to fix this problem? If we really do need it, we should refactor the code used to compute the datatype label color and use the same code (via mixin)

Reply:
I put make_color in a mixin
Also it's not that setting the color was the problem, it was the use of datatypetag. I think it's a Vue bug, sometimes the change event for datatype as a prop to datatypetag wouldn't get called since it was templatized